Backpack-SwiftUI/ChipGroup/Classes/BPKSingleSelectChipGroup.swift

```diff
@@ -0,0 +1,146 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A Group of chips that allows a single chip to be selected at a time.
+public struct BPKSingleSelectChipGroup: View {
+    private let chips: [ChipItem]
+    private let style: BPKChipStyle
+    private let type: BPKChipGroupType
+    private let onItemClick: (_ index: Int) -> Void
+    
+    @Binding private var selectedIndex: Int?
+    
+    public init(
+        chips: [ChipItem],
+        style: BPKChipStyle = .default,
+        selectedIndex: Binding<Int?>,
+        type: BPKChipGroupType = .rail,
+        onItemClick: @escaping (_ index: Int) -> Void
+    ) {
+        self.chips = chips
+        self.style = style
+        _selectedIndex = selectedIndex
+        self.type = type
+        self.onItemClick = onItemClick
+    }
+    
+    public var body: some View {
+        switch type {
+        case .rail:
+            ScrollView(.horizontal, showsIndicators: false) {
+                HStack(spacing: .md) {
+                    chipsListView
+                }
+            }
+        case .wrap:
+            FlowStackView(spacing: .init(width: .md, height: .md)) {
+                chipsListView
+            }
+        }
+    }
+    
+    private var chipsListView: some View {
+        ForEach(0..<chips.count, id: \.self) { index in
+            BPKChip(
+                chips[index].text,
+                icon: chips[index].icon,
+                selected: selectedIndex == index,
+                style: style) {
+                    selectedIndex = index
+                    onItemClick(index)
+                }
+        }
```

Comment:
I removed the ChipItemView created before, as it wasn't doing anything really, here I'm just using the BPKChip directly. Les code to maintain! :D